dot/network/service.go

```diff
@@ -280,6 +282,40 @@ func (s *Service) logPeerCount() {
 func (s *Service) handleConn(conn libp2pnetwork.Conn) {
 	// give new peers a slight weight
 	s.syncQueue.updatePeerScore(conn.RemotePeer(), 1)
+
+	s.notificationsMu.Lock()
+	defer s.notificationsMu.Unlock()
+
+	if _, has := s.notificationsProtocols[BlockAnnounceMsgType]; !has {
+		// this shouldn't happen
+		logger.Warn("block announce protocol is not yet registered!")
+		return
+	}
+
+	info := s.notificationsProtocols[BlockAnnounceMsgType]
+
+	// open block announce substream
+	hs, err := info.getHandshake()
+	if err != nil {
+		logger.Warn("failed to get handshake", "protocol", blockAnnounceID, "error", err)
+		return
+	}
+
+	info.mapMu.RLock()
+	defer info.mapMu.RUnlock()
+
+	peer := conn.RemotePeer()
+	if hsData, has := info.handshakeData[peer]; !has || !hsData.received {
+		info.handshakeData[peer] = &handshakeData{
+			validated: false,
```

Comment:
Should we set also set `received` to `true` or reuse the `hsData`?

Reply:
no, at this point it hasn't sent or received any handshakes